wgpu-core/src/identity.rs

```diff
@@ -48,28 +51,32 @@ impl IdentityValues {
     /// The backend is incorporated into the id, so that ids allocated with
     /// different `backend` values are always distinct.
     pub fn alloc<T: Marker>(&mut self, backend: Backend) -> Id<T> {
+        assert!(
+            self.allocate_ids != Some(false),
+            "Mix of internally allocated and externally provided IDs"
+        );
+        self.allocate_ids = Some(true);
```

Comment:
```suggestion
        assert_eq!(
            self.allocate_ids.get_or_insert(true),
            true,
            "Mix of internally allocated and externally provided IDs"
        );
```

Another form which feels personally cleaner.